pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1202,6 +1211,22 @@ type Sigv4 struct {
 	RoleArn string `json:"roleArn,omitempty"`
 }
 
+// AzureAD defines the configuration for remote write's azuread parameters.
+// +k8s:openapi-gen=true
+type AzureAD struct {
+	// The Azure Cloud. Options are 'AzurePublic', 'AzureChina', or 'AzureGovernment'. If blank, AzurePublic is used.
+	Cloud string `json:"cloud,omitempty"`
```

Comment:
I think would be useful have this annotation `// +kubebuilder:validation:Enum=AzureChina; AzureGovernment; AzurePublic"` and maybe `// +kubebuilder:default:="AzurePublic"`

Reply:
agreed on this. I also changed the field comment to `// The Azure Cloud. Options are 'AzurePublic', 'AzureChina', or 'AzureGovernment'.` 😄

Reply:
Is this field really optional?

Could we remove the `// +optional` comment if it isn't, or change the type to `*string` if it really is optional?

Reply:
https://github.com/prometheus/prometheus/blob/main/docs/configuration/configuration.md?plain=1#L3536, yup I think that is optional and I change the type to `*string` 😄